Add policy for adding new maintainers (#434)
space-wizards/docs: src/en/wizden-staff/maintainer/adding-new-maintainers.md

# Adding New Maintainers

> This document outlines the process for adding new maintainers to Wizden, it is intended for the current maintainers and project managers to follow when adding new maintainers to the project. 

Reviewer: ```suggestion
This document outlines the process for adding new maintainers to Wizden.
```

feels redundant to specify that the process is meant to be followed.

> This document outlines the process for adding new maintainers to Wizden, it is intended for the current maintainers and project managers to follow when adding new maintainers to the project. 

Reviewer 2: Agreed. Otherwise this is a run-on sentence.

## Requirements

> As maintainers represent wizden as staff, certain requirements must be met before a person can be a maintainer. These requirements are as follows:

Reviewer: ```suggestion
As maintainers represent Wizden as staff, certain requirements must be met before a person can be a maintainer. These requirements are as follows:
```
capitalization

> As maintainers represent wizden as staff, certain requirements must be met before a person can be a maintainer. These requirements are as follows:

Reviewer: ```suggestion
As maintainers represent wizden as staff, certain requirements must be met before a person can be a maintainer:
```
Reads a bit cleaner and is shorter.

- Be a member of the Space Station 14 Discord server, as many discussions happen there.
- Not have had any singificant administrative issues in the past, such as being banned from the game or Discord server or having a history of being rude to other members of the community.
    - This is a subjective requirement, and should be discussed in the voting process.
- Have a good understanding of the codebase and how it works.

>     - This is a subjective requirement, and should be discussed in the voting process.

Reviewer 2: ```suggestion
    - This is a subjective requirement, and should be discussed in the voting process.
 - Have a solid understanding the culture and values of the game and community
```

### Voting Process

> Before a vote can start, the person being voted on must be nominated by a current maintainer. Once a person has been nominated, the following process should be followed:

Reviewer 3: This reads a bit strange and confusing to me, it kinda implies the existence of a state where the candidate is being considered, but vote can't be started until they get nominated... Which is actualy how their name comes up in the first place, of course.
This seems more straightforward to me:

```suggestion
To add a new maintainer, a current maintainer has to nominate the candidate. Once a person has been nominated, the following process should be followed:
```

1. The person being nominated should be contacted and asked if they would like to be nominated. If they agree, the nomination can proceed.
2. A vote should be created "Internal" category on Discourse, using the Discourse template.
3. The vote should be open for 1 week, and all current maintainers should be notified of the vote. During this time, maintainers should discuss the nomination and ask any questions they may have. If there is a clear majority in favor of the nomination, the vote can be closed early. If there is a clear majority against the nomination, the vote should be closed and the nomination should be withdrawn.
    - If there is no clear majority, the vote should remain open for the full week.
    - If there are any concerns about the nomination, they should be discussed in the thread and addressed before the vote is closed.
4. At the end of the week, the vote should be closed and the results should be announced. If the vote passes, the person being nominated should be contacted and asked if they would like to accept the position. If they agree, they can be added to the maintainers list.

> ### Special Maintainer Roles

Reviewer: this bullet points explaining the additional maintainer roles are unnecessary. You could remove all of this below and replace it with something like this:

For adding maintainers with special duties (engine maintainers, art leads, etc.), the process for voting is the same as above but with the following changes:

Author: I think it's good if it's well defined. Saying "etc." leaves it open to interpretation.

Some roles are special Maintainer roles and follow a different process to the normal Maintainer role. These roles are as follows:
- **Head Mapper**: A Maintainer responsible for ensuring mapping standards and approving and merging mapping PRs.
- **Art lead**: A Maintainer responsible for ensuring a consistent art style and approving sprite PRs.
- **Robust Toolbox Maintainer**: A Maintainer of Robust Toolbox, our underlying game engine. 

> 

Reviewer 4: I'd probably remove this as I think RT maintainers would entirely be under PJB's purview unless that were no longer feasible, in which case we would add them under the policy.

#### Starting a Vote for a Special Role

> The process for adding a special maintainer is the same as above, but with the following changes:
> - Only a member that is already a part of the special role can nominate a new member for that role.
>     - The only exception is when the role is empty, in which case any maintainer can nominate a new member.
> - The vote will be open for 2 weeks instead of 1 week.

Reviewer 5: ```suggestion
The process for adding a special Maintainer is the same as for a normal Maintainer, but with the following alterations:
1. If a special role has at least one member, only a member that is already a part of the special role can nominate a new member for that role.
2. Votes to add a special Maintainer will be open for 2 weeks instead of 1 week.
```

## Requirements

Maintainers represent Space Wizards as staff. Certain requirements must be met before a person can become a Maintainer:
- A vote must be held according to the Voting Process outlined below.
- Be a member of the Space Station 14 Discord server.
- If the nominee has caused Administration issues in the past, these issues must be raised and discussed during the voting process. Administration issues include, but are not limited to, being banned from the game, banned from the Discord server, or having a history of being rude to other members of the community.
- Have a good understanding of the Space Station 14 codebase and how it works. This must be raised and discussed during the voting process.
- Have a solid understanding of the culture and values of the game and community

### Voting Process

To add a new Maintainer, a current Maintainer has to nominate the candidate. Once a person has been nominated, the following process should be followed:
1. A vote will be created in the "Internal" category on the Space Station 14 Discourse, using the appropriate template. This vote will be open for one week, and all Maintainers will be notified of the vote.

> 2. During this time, Maintainers should discuss the nomination, raise concerns, and ask any questions they may have. If there is a clear majority of Maintainers in favor or against the nomination, the vote may be closed early.

Reviewer 3: The only mild concern I have left is that since the "close the vote early" part is left rather vague on the specifics, someone might feel aggrieved if the vote process was called before they could raise some concern they had, and then they could feel wronged. Probably not likely to ever come up, but, uh, if it ever does we will sure say "this should have been more specific"? What do you think

Author: Clear Majority in this case would be something like "It's been a day and 10 Maintainers said yes and no one said no"

Reviewer 3: If this is meant to be for unanimous (preliminary) results, should it not say that?

Author: It should, I'll amend it.

- Have a good understanding of the Space Station 14 codebase and how it works. This must be raised and discussed during the voting process.
- Have a solid understanding of the culture and values of the game and community

### Voting Process

To add a new Maintainer, a current Maintainer has to nominate the candidate. Once a person has been nominated, the following process should be followed:
1. A vote will be created in the "Internal" category on the Space Station 14 Discourse, using the appropriate template. This vote will be open for one week, and all Maintainers will be notified of the vote.
2. During this time, Maintainers should discuss the nomination, raise concerns, and ask any questions they may have. If there is a clear majority of Maintainers in favor or against the nomination, the vote may be closed early.
    - If there is no clear majority, the vote must remain open for the full week.
    - Concerns about a nominated person must be addressed before the vote may close. This will delay the vote until the concerns are resolved.
    - For a vote to pass, it must have a supermajority (66%) of votes in favor. 

>         - Abstain votes are considered neutral. However, if counting all abstain votes as "No" would cause the vote to fall below the required 66% threshold, the vote fails.

Reviewer: I dont understand how its an "abstain" if it still counts for the voting and can cause the vote to fail. Voting to abstain conventionally just means that you are formally not voting for either side, and i see no reason why it should be different here.
